File: spring/chatbox/chatter/src/main/java/chatter/service/MessageService.java
package chatter.service;

import java.util.List;

import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import chatter.entity.Message;


public interface MessageService {
	public List<Message> getMessageBySenderId(int userId, int receiverid);
	public List<Message> getMessageByReceiverId(int userId, int receiverid);
	public void setMessage(Message msg);
}

File: Angular/chatbox/src/app/services/message.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Message } from '../entity/Message';
import { User } from '../entity/User';

@Injectable({
  providedIn: 'root'
})
export class MessageService {

  constructor(private http : HttpClient){}
  restUrl :  string = "http://localhost:8080/chatter/";
  httpOptions = {
    headers : new HttpHeaders({
      'content-Type' : 'application/json'
    })
  }

  



  getMessageBySenderId(senderid : any,receiverid: any):Observable<Message[]>{
    return this.http.get<any>(this.restUrl+"msgbysenderid/"+senderid+"/"+receiverid );
  }


  // getMessageByReceiverId(userid : any):Observable<Message[]>{
  //   return this.http.get<any>(this.restUrl+"msgbyreceiverid/"+userid );
  // }

  setMessage(message : any){
    return this.http.post<any>(this.restUrl+'setmessage', JSON.stringify(message), this.httpOptions);
  }
}
